### PR TITLE
phenix.refine compatibility, find_cra() for connection partners, typo (triangles), factor, importable script

### PR DESCRIPTION
**Pull request by MartinMalyMM**
Ready to merge.

restructured: can be used as a script from command line as well as imported in another python script

### CODE REVIEW

**Comment by MartinMalyMM** [comment]
Please wait before merging this, I will make another correction next week.

**Comment by MartinMalyMM** [comment]
The pull request should be now ready for merging.

File: tutorial/json2restraints.py [hunk 1]
```diff
@@ -47,19 +47,23 @@
...
         connections_kept = gemmi.ConnectionList()
         for connection in st.connections:
-            if not (connection.type == gemmi.ConnectionType.MetalC or \
-                    connection.partner1.atom.element.is_metal or \
-                    connection.partner2.atom.element.is_metal):
-                connections_kept.append(connection)
+            partner1_atom = st[0].find_cra(connection.partner1, ignore_segment=True).atom
+            partner2_atom = st[0].find_cra(connection.partner2, ignore_segment=True).atom
+            if partner1_atom and partner2_atom:  # check if atoms exist
+                if not (connection.type == gemmi.ConnectionType.MetalC or \
```
Comment (keitaroyam) [inline tutorial/json2restraints.py]: Probably we don't need to test `connection.type == gemmi.ConnectionType.MetalC`. The connection.type may be even wrongly set.

File: tutorial/json2restraints.py [hunk 2]
```diff
@@ -47,19 +47,23 @@
 
 outputRestraintsPath = outputPrefix + ".txt"
 outputRestraintsCootPath = outputPrefix + "_coot.txt"
+outputRestraintsPhenixPath = outputPrefix + ".def"
```
Comment (keitaroyam) [inline tutorial/json2restraints.py]: not .params?

File: tutorial/json2restraints.py [hunk 3]
```diff
@@ -57,13 +57,13 @@
...
         connections_kept = gemmi.ConnectionList()
         for connection in st.connections:
-            partner1_atom = st[0].find_cra(connection.partner1, ignore_segment=True).atom
-            partner2_atom = st[0].find_cra(connection.partner2, ignore_segment=True).atom
-            if partner1_atom and partner2_atom:  # check if atoms exist
-                if not (connection.type == gemmi.ConnectionType.MetalC or \
-                        partner1_atom.element.is_metal or \
-                        partner2_atom.element.is_metal):
-                    connections_kept.append(connection)
+            for m in range(len(st)):
```
Comment (keitaroyam) [inline tutorial/json2restraints.py]: This is ok, but you could simply `for model in st` unless you need indices

**Comment by MartinMalyMM** [comment]
The pull request is now ready for merging.

**Pull request merged**